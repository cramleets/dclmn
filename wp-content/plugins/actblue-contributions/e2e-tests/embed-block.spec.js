import {
	createNewPost,
	enablePageDialogAccept,
	getEditedPostContent,
	insertBlock,
} from "@wordpress/e2e-test-utils";

import { selectBlockByName } from "./helpers";

jest.setTimeout(15000);

describe("Wrapper block", () => {
	beforeAll(async () => {
		await enablePageDialogAccept();
	});
	beforeEach(async () => {
		await createNewPost();
	});

	// Tests can be added here by using the it() function

	it("Embed block is available", async () => {
		await insertBlock("ActBlue Embed");

		// Check if block was inserted
		expect(await page.$('[data-type="actblue/embed"]')).not.toBeNull();

		expect(await getEditedPostContent()).toMatchSnapshot();
	});

	it("Add embed url", async () => {
		await insertBlock("ActBlue Embed");
		await selectBlockByName("actblue/embed");

		// Type in the embed URL.
		await page.type(
			'input[aria-label="ActBlue URL"]',
			"https://secure.actblue.com/donate/actblue-1-embed"
		);

		// Click the "Embed" button.
		const [embedBtn] = await page.$x(
			'//div[@data-type="actblue/embed"]//button[contains(text(), "Embed")]'
		);
		await embedBtn.click();

		// wait for 1 second for the embed to load.
		await page.waitForTimeout(1000);

		expect(await getEditedPostContent()).toMatchSnapshot();
	});
});
