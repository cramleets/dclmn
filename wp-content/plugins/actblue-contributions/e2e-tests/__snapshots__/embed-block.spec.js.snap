// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper block Add embed url 1`] = `
"<!-- wp:actblue/embed {\\"url\\":\\"https://secure.actblue.com/donate/actblue-1-embed\\"} -->
<figure class=\\"wp-block-actblue-embed wp-block-embed is-provider-actblue\\"><div class=\\"wp-block-embed__wrapper\\">
https://secure.actblue.com/donate/actblue-1-embed
</div></figure>
<!-- /wp:actblue/embed -->"
`;

exports[`Wrapper block Embed block is available 1`] = `"<!-- wp:actblue/embed /-->"`;
